src/Nelmio/Alice/Fixtures/Parser/Methods/Base.php

```diff
@@ -102,17 +102,18 @@ protected function processIncludes($data, $filename)
     /**
      * @param array $data
      * @param array $includeData
+     * @return array
      */
     protected function mergeIncludeData($data, $includeData)
     {
-        foreach ($includeData as $class => $fixtures) {
-            if (isset($data[$class])) {
-                $data[$class] = array_merge($fixtures, $data[$class]);
+        foreach ($data as $class => $fixtures) {
+            if (isset($includeData[$class])) {
+                $includeData[$class] = array_merge($includeData[$class], $fixtures);
             } else {
-                $data[$class] = $fixtures;
+                $includeData[$class] = $fixtures;
             }
         }
 
-        return $data;
+        return $includeData;
```

Comment:
@theofidry i created $newData instead of reusing $includeData just for code clarity, as stated here by @tshelburne 
If you think it is incorrect, there's no problem, we can rollback whenever we want, of course :)